src/ui/settings/ViewSubscriptionManagement/ViewSubscriptionManagement.qml

```diff
@@ -0,0 +1,221 @@
+                VPNSubscriptionData.planIntervalCount
+            ),
+            type: "text",
+        });
+
+        subscriptionInfoModel.append({
+            labelText: VPNl18n.SubscriptionManagementStatusLabel,
+            valueText: VPNSubscriptionData.status,
+            type: "pill",
+        });
+
+        subscriptionInfoModel.append({
+            labelText: VPNl18n.SubscriptionManagementNextLabel,
+            // TODO: Calculate next billing date if any
+            valueText: "",
+            type: "text",
+        });
+
+        subscriptionInfoModel.append({
+            labelText: VPNl18n.SubscriptionManagementActivatedLabel,
+            valueText: unixToDate(VPNSubscriptionData.createdAt),
+            type: "text",
+        });
+
+        subscriptionInfoModel.append({
+            labelText: VPNl18n.SubscriptionManagementExpiresLabel,
+            valueText: unixToDate(VPNSubscriptionData.expiresOn),
+            type: "text",
+        });
+
+        // Subscription payment model
+        if (VPNSubscriptionData.paymentType === "credit") {
+            subscriptionPaymentModel.append({
+                labelText: VPNSubscriptionData.creditCardBrand,
+                valueText: VPNl18n.SubscriptionManagementCardLast4.arg(VPNSubscriptionData.creditCardLast4),
+                type: "creditCard",
+            });
+
+            subscriptionPaymentModel.append({
+                labelText: VPNl18n.SubscriptionManagementCardExpiresLabel,
+                // TODO: Show date w/o days
+                valueText: unixToDate(new Date(VPNSubscriptionData.creditCardExpYear, VPNSubscriptionData.creditCardExpMonth - 1).getTime() / 1000),
+                type: "text",
+            });
+        } else {
+            subscriptionPaymentModel.append({
+                // TODO: Handle different IAP platforms
+                labelText: VPNl18n.SubscriptionManagementSubscriptionPlatformLabel.arg("IAP Google or Apple"),
+                valueText: "",
+                type: "text",
+            });
+        }
+    }
+
+    function unixToDate(unixTimestamp) {
+        return new Date(unixTimestamp * 1000).toLocaleDateString(currentLocale, Locale.ShortFormat);
+    }
+
+    function getPlanText(currencyCode, amount, intervalCount) {
+        const amountDisplay = (amount || 0) / 100;
+        const localizedCurrency = VPNLocalizer.localizeCurrency(amountDisplay, currencyCode.toUpperCase());
+
+        if (intervalCount === 12) {
+            // {¤amount} yearly
+            return VPNl18n.SubscriptionManagementPlanValueYearly.arg(localizedCurrency);
+        } else if (intervalCount === 6) {
+            // {¤amount} half-yearly
+            return VPNl18n.SubscriptionManagementPlanValueHalfYearly.arg(localizedCurrency);
+        } else if (intervalCount === 1) {
+            // {¤amount} monthly
+            return VPNl18n.SubscriptionManagementPlanValueMonthly.arg(localizedCurrency);
+        }
```

Comment:
Can you write an warning message + a glean event if we are here (and it's not a free-trial)? Because maybe tomorrow we will introduce a new `intervalCount` and we could forget to update this view.

Reply:
Yes, good idea!